### ISSUE TITLE
Custom heatmap which interpolates the points

### ISSUE THREAD

**Issue by alimirzaei** [visgl/deck.gl#4827]
I want to change the heat map layer which interpolates point weights with a specific radius. In the current version, the intensity of each location is calculated based on sum of intensity of near weighted points. I want to have the average value (not the sum). I want to implement interpolation methods like IDW with specific maxRadius. How can I change the current layer to have this layer? I will appreciate you, if you can help me.

**Comment by robinator8** [comment]
If you are looking for the creation of heat map that displays continuous values instead of point density then I am in agreement with you. It would also be necessary to have the colors fade in opacity but not change in color. 

Temperature data is a perfect example: currently if you have a single data point, it will appear that exactly at the point there is a high temperature, and all around the data point the temperature decreases. This is of course not what you're looking for in this situation.

I am looking for a heatmap that can properly display that kind of continuous data (e.g. temperature) but haven't been able to get the heatmap implementation to work.

TLDR: +1 on the question (if I understand the question correctly)

**Comment by Pessimistress** [comment]
The `HeatmapLayer` is designed to handle discrete positions. For continuous data you want to create a mesh with something like [d3-delaunay](https://github.com/d3/d3-delaunay) and render with `SimpleMeshLayer` (which will also give you interpolation for free).

**Comment by Pessimistress** [comment]
An example layer that uses Delaunay triangulation: https://github.com/visgl/deck.gl/tree/x/delaunay/examples/experimental/delaunay

**Comment by Pessimistress** [comment]
You can also try `8.4.0-alpha.1` with the new `aggregation` prop:

```js
new HeatmapLayer({
  ...
  aggregation: 'MEAN' // 'SUM' or 'MEAN'
})
```
